components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/servlet/SAMLSSOProviderServlet.java

```diff
@@ -281,6 +287,129 @@ private void handleRequest(HttpServletRequest req, HttpServletResponse resp, boo
         }
     }
 
+    private void handleInvalidRequestMessage(HttpServletRequest req, HttpServletResponse resp, String sessionId)
+            throws IOException, IdentityException, ServletException {
+
+        log.debug("Invalid request message or single logout message ");
+
+        if (sessionId == null) {
+            String errorResp = SAMLSSOUtil.buildErrorResponse(
+                    SAMLSSOConstants.StatusCodes.REQUESTOR_ERROR,
+                    "Invalid request message", null);
+            sendNotification(errorResp, SAMLSSOConstants.Notification.INVALID_MESSAGE_STATUS,
+                    SAMLSSOConstants.Notification.INVALID_MESSAGE_MESSAGE, null, req, resp);
+        } else {
+            // Non-SAML request are assumed to be logout requests
+            sendToFrameworkForLogout(req, resp, null, null, sessionId, true,
+                    false);
+        }
+    }
+
+    private void handleSAMLResponse(HttpServletRequest req, HttpServletResponse resp, String samlResponse,
+                                    String sessionId)
+            throws IdentityException, IOException, ServletException {
+
+        XMLObject response = SAMLSSOUtil.unmarshall(SAMLSSOUtil.decode(samlResponse));
+
+        if (!(response instanceof LogoutResponseImpl)) {
+            handleInvalidRequestMessage(req, resp, sessionId);
+        } else {
+            String inResponseToId = ((LogoutResponseImpl) response).getInResponseTo();
```

Comment:
We need to call FrontChannelSLOParticipantInfo frontChannelSLOParticipantInfo =
getFrontChannelSLOParticipantInfo(inResponseToId); here. If frontChannelSLOParticipantInfo == null, call handleInvalidRequestMessage(req, resp, sessionId);